Sources_v3/Controllers/ChannelListController/ChannelListController.swift

```diff
@@ -94,29 +95,22 @@ public class ChannelListControllerGeneric<ExtraData: ExtraDataTypes>: DataContro
         self.environment = environment
     }
     
-    /// Starts updating the results.
+    /// Synchronize local data with remote.
     ///
-    /// 1. **Synchronously** loads the data for the referenced objects from the local cache. These data are immediately available in
-    /// the `channels` property of the controller once this method returns. Any further changes to the data are communicated
-    /// using `delegate`.
+    /// 1. **Synchronously** loads the data for the referenced objects from the local cache if it is not loaded yet.
+    /// Any further changes to the data are communicated using `delegate`.
     ///
-    /// 2. It also **asynchronously** fetches the latest version of the data from the servers. Once the remote fetch is completed,
+    /// 2. It also **Asynchronously** fetches the latest version of the data from the servers. Once the remote fetch is completed,
     /// the completion block is called. If the updated data differ from the locally cached ones, the controller uses the `delegate`
     /// methods to inform about the changes.
     ///
     /// - Parameter completion: Called when the controller has finished fetching remote data.
     ///                         If the data fetching fails, the `error` variable contains more details about the problem.
-    public func startUpdating(_ completion: ((_ error: Error?) -> Void)? = nil) {
-        do {
-            try channelListObserver.startObserving()
-        } catch {
-            log.error("Failed to perform fetch request with error: \(error). This is an internal error.")
-            callback { completion?(ClientError.FetchFailed()) }
-            return
-        }
-
-        // Update observing state
-        state = .localDataFetched
+    public func synchronize(_ completion: ((_ error: Error?) -> Void)? = nil) {
```

Comment:
I thought we were going to implicitly init `channelListObserver` in `init` (so local data is available) but `synchronize` will only be used for remote data?

Reply:
Yep, `synchronize()` will be for syncing remote data, but if we will fetch local data on each init, it can become rather expensive initializing.
So after discussion with @VojtaStavik we ended up on having observers ready in following cases:
- Property access. (e.g. after accessing channel, you will get this channel and observing will start)
- Setting delegate. (After creating controller and setting delegate you will be able to receive updates)
- (?) Also previously I added it also to `synchronize()` call, but now it seems like we don't have use case for that - if no delegate is set (Combine and SwiftUI are also delegates in this case) then there is no need to observe changes.

Reply:
Removed DB observing trigger from `synchronize()`.